### PR TITLE
fix: mail to links replace ampersand [TOL-923]

### PR DESCRIPTION
**Pull request by YvesRijckaert**
We use `dompurify` to sanitize the markdown preview output. Dompurify uses innerHTML at some point, which transforms `&` to`&amp` inside of href links. This causes an issue with mailto links not working as expected. Everything after `&amp` gets ignored.

For example: `<a href="mailto:test@test.com?subject=test&body=test">Test</a>`
turns into:
`<a href="mailto:test@test.com?subject=test&amp;body=test">Test</a>`, which makes the `mailto` link ignore the body parameter.

I added a regex that transforms the `&amp` to `&` again inside mail to links. Unfortunately I was not able to add any exclusion inside the dompurify package (see this issue in the package: https://github.com/cure53/DOMPurify/issues/379#issuecomment-571997013)

### CODE REVIEW

**Review by aodhagan-cf — APPROVED** [review]
Looks good! 👍

**Review by z0al — CHANGES_REQUESTED** [review]
Sorry for blocking but this gets modified by `DOMPurify` to avoid security issues. I don't recommend altering the behavior this way. 

Either we continue to sanitize the attribute and communicate that in our docs or we add an exclusion rule for the `mailto` attribute (see [here](https://github.com/cure53/DOMPurify)). Please consult our security team

**Comment by YvesRijckaert** [comment]
> Sorry for blocking but this gets modified by `DOMPurify` to avoid security issues. I don't recommend altering the behavior this way.
> 
> Either we continue to sanitize the attribute and communicate that in our docs or we add an exclusion rule for the `mailto` attribute (see [here](https://github.com/cure53/DOMPurify)). Please consult our security team

I will contact the security team for a review. The exclusion does not work since dompurify uses innerHtml under the hood, which transforms the `&` (see the linked issue)
Let's see what the security team says, if we cannot ship this change, let's go for option 1.

**Comment by roryscarson** [comment]
@z0al Can you explain the concern to me? DOMPurify is still being used on the initial input and the new replace function only accepts the input string, not a value to replace, so it can't be as easily manipulated. This seems fine to me but I could be missing something.

**Comment by z0al** [comment]
> @z0al Can you explain the concern to me? DOMPurify is still being used on the initial input and the new replace function only accepts the input string, not a value to replace, so it can't be as easily manipulated. This seems fine to me but I could be missing something.

The idea behind DOMPurify is to sanitize the markdown output we are previewing within Contentful web app to avoid XSS. With this change, we are taking the sanitized output and changing it again undoing some of the sanitization done by the library. 

I'm not a subject matter expert, and not sure if this is will be enough to open a security threat but it raised a flag for me. I'm fine with going with the security team's recommendation.

**Review by roryscarson — APPROVED** [review]

File: packages/markdown/src/components/MarkdownPreview.tsx
Comment (karankohli-cf) [inline packages/markdown/src/components/MarkdownPreview.tsx:205]: might want to look at 
https://github.com/cure53/DOMPurify/blob/f5c3e0b3f8c17b4d3fa0a827038f0a91e2e244ad/README.md?plain=1#L280

Comment (YvesRijckaert) [reply packages/markdown/src/components/MarkdownPreview.tsx:205]: Unfortunately this did not fix the issue

**Pull request merged**
